lib/tests/systems.nix

```diff
@@ -102,17 +102,22 @@ lib.runTests (
         arbitraryValue = x: "<<modified>>";
     in {
       expr = lib.systems.equals (lib.systems.elaborate "x86_64-linux") modified;
-      expected = {
-        # Changes in these attrs are not detectable because they're function.
-        # The functions should be derived from the data, so this is not a problem.
-        canExecute = null;
-        emulator = null;
-        emulatorAvailable = null;
-        staticEmulatorAvailable = null;
-        isCompatible = null;
-      }?${platformAttrName};
+      expected = false;
     };
-
-}) (lib.systems.elaborate "x86_64-linux" /* arbitrary choice, just to get all the elaborated attrNames */)
+  })
+  # x86_64-linux is an arbitrary choice, just to get all the elaborated attrNames
+  (builtins.removeAttrs (lib.systems.elaborate "x86_64-linux") [
+    # Avoid the warning
+    "warnAboutPlatformEquals"
+    # Changes in these attrs are not detectable because they're function.
+    # The functions should be derived from the data, so this is not a problem.
+    "canExecute"
+    "emulator"
+    "emulatorAvailable"
+    "equals"
+    "isCompatible"
+    "notEquals"
+    "staticEmulatorAvailable"
+  ])
```

Comment:
This PR was adding a `removeAttrs` here anyway, so I've added an extra commit that refactors this test to take advantage of it.

This is a little out of scope, so I can drop the commit if preferred.